### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by jburel**
Replace travis by GHA

cc @sbesson

### CODE REVIEW

**Comment by jburel** [comment]
see https://github.com/jburel/bio-formats-octave-docker/actions/runs/403071520

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v2
+      - name: Build
+        run: |
+          docker build -t bio-formats-octave .
+      - name: Test
+        shell: bash
+        run: |
+          mkdir data
+          touch "data/test&sizeT=3.fake"
+          if [ ! -f data/tubhiswt-2D/tubhiswt_C0.ome.tif ]; then wget --user-agent GHA https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/tubhiswt-2D.zip && unzip tubhiswt-2D.zip && mv tubhiswt-2D data; fi
```
Comment (sbesson) [inline .github/workflows/main.yml]: The first conditional only makes sense if we introduce some caching for the downloaded data.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build
+        run: |
+          docker build -t bio-formats-octave .
+      - name: Test
```
Comment (sbesson) [inline .github/workflows/main.yml]: Maybe separate the step downloading/creating the sample data from the step running the tests?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,6 @@
 Bio-Formats Octave
 ==================
 
-[![Build Status](https://travis-ci.org/openmicroscopy/bio-formats-octave-docker.svg)](https://travis-ci.org/openmicroscopy/bio-formats-octave-docker)
```
Comment (sbesson) [inline README.md:4]: Need updating to point at the actions

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        shell: bash
+        run: |
+          mkdir data
+          touch "data/test&sizeT=3.fake"
+          if [ ! -f data/tubhiswt-2D/tubhiswt_C0.ome.tif ]; then wget --user-agent GHA https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/tubhiswt-2D.zip && unzip tubhiswt-2D.zip && mv tubhiswt-2D data; fi
+          docker run -t -v $(pwd)/data/:/data/ bio-formats-octave ci_test.m /data/tubhiswt-2D/tubhiswt_C0.ome.tif
```
Comment (sbesson) [inline .github/workflows/main.yml]: The renaming of the file to drop Travis makes sense. Maybe it should even be simply `test.m` (like `test.sh` in other Docker repos)

**Review by sbesson — CHANGES_REQUESTED** [review]
A few suggestions/comments.

Independent of the changes proposing in PR, while reviewing the various Docker repositories and the migration to GitHub actions, I feel there might be an opportunity to rationalize/automate the  Docker images release workflow (currently based on `Makefile` copied in various repos). I suppose you don't have anything staging that would achieve this functionality? /cc @joshmoore

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build
+        run: |
+          docker build -t bio-formats-octave .
+      - name: Prepare data
+        run: |
+          mkdir data
+          touch "data/test&sizeT=3.fake"
+          if [ ! -f data/tubhiswt-2D/tubhiswt_C0.ome.tif ]; then wget --user-agent GHA https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/tubhiswt-2D.zip && unzip tubhiswt-2D.zip && mv tubhiswt-2D data; fi
```
Comment (sbesson) [inline .github/workflows/main.yml]:
```suggestion
          wget --user-agent GHA https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/tubhiswt-2D.zip && unzip tubhiswt-2D.zip && mv tubhiswt-2D data
```

assuming the caching is a next todo

**Review by sbesson — APPROVED** [review]

**Comment by sbesson** [comment]
:+1: as my question about the release workflow was unanswered, I assume we will come back to it in a second step. Happy to see this merged immediately to continue the transition to GHA

**Pull request merged**